lib/response.js

```diff
@@ -84,7 +84,7 @@ module.exports = {
     if (this.headerSent) return;
 
     assert('number' == typeof code, 'status code must be a number');
-    assert(statuses[code], `invalid status code: ${code}`);
+    assert(/^[0-9]{3}$/.test(code), `invalid status code: ${code}`);
```

Comment:
how about 

```js
assert(Number.isIngeter(code) && code >= 100 && code <= 999, 'message')
```

Reply:
Yea, I'll update it. I ran a quick bench just to check how bad RegExp is in comparison (even though it's pointless in a real world scenario as I can't imagine `ctx.status` being a bottleneck:

```sh
Valid RegExp x 37,061,865 ops/sec ±0.51% (88 runs sampled)
Valid expression x 728,862,990 ops/sec ±1.60% (82 runs sampled)
```

The only thing I'd change is to not check both for number and Number.isInteger. I'll move isInteger to the above assert and keep that message (less "breaking") and replace RegExp.